### ISSUE TITLE
.xoj mime-type does not get installed, when installing xournal via nix-env.

### ISSUE THREAD

**Issue by maralorn** [NixOS/nixpkgs#31328]
## Issue description

The mime-type file included in the xournal package in merge request #20558 does not get linked into the user-environment when installed with nix-env. This works, when installed via system environment. I cannot tell, if this is a problem with nix or this specific package. Or is this even intended? I don’t know enough about mime-types in nixos to fix this on my own.

### Steps to reproduce

1. Add `xournal` to `environment.systemPackages` and rebuild.
2. Run `nix-env -iA nixos.xournal`.
3. Compare the output off `find /nix/store -name 'xournal.xml'` (Do `nix-collect-garbage` first, if the output is very long. There probably is a better way to do this, please enlighten me.):

I find a line

    /nix/store/…-system-path/share/mime/packages/xournal.xml

but no line 

    /nix/store/…-user-environment/share/mime/packages/xournal.xml

## Technical details

* System: `Nixos 17.09.1997.1a8a95e879 (Hummingbird)`
* Nix version: `1.11.15`
* Sandboxing enabled: no

**Comment by maralorn** [comment]
Also, once I installed `xournal` via the system-config now the mime-type get's discovered correctly but my gnome does not find xournal although it found the desktop-file and shows xournal in the application list. No clue, what’s wrong there … But I have not debugged it enough to file an issue.

**Comment by teto** [comment]
There is the same problem for manpage. nix-env doesn't do all the magic.

**Comment by maralorn** [comment]
@teto Do you refer to xournal specifically?

**Comment by teto** [comment]
no, it s a generic problem, i suppose both are linked

**Comment by matthewbauer** [comment]
@maralorn Are you sure it's not being installed?

The location after ```nix-env -iA``` should be in ```~/.nix-profile/share/mime/packagess```. This works for me at least. You might be getting confused by some of the symlinks.

Now for your DE to actually recognize it might be a little more difficult... You probably would would have to generate the database yourself with something like:

```
XDG_DATA_DIRS=~/.nix-profile/share/mime update-mime-database -V ~/.nix-profile/share/mime
```

That's probably not going to work out of the box but with something like ```buildEnv``` it's definitely possible. Anyway this is something that NixUP(#9250) will hopefully fix eventually.

**Comment by garbas** [comment]
I think @matthewbauer answered as best as nix can do at the moment. Since this is a more general problem which - hopefully - is going to be addressed by #9250.

@maralorn, feel free to reopen the ticket if you disagree.

**Comment by danbst** [comment]
If you want for this to work with regular `nix-env` (without home-manager or any other declarative env approach), you may add some hacks https://github.com/NixOS/nixpkgs/issues/13134#issuecomment-567880990
